src/compressed_tensors/transform/factory/hadamard.py

```diff
@@ -54,13 +53,21 @@ def create_transform(self, module: Module, args: TransformArgs):
         size = get_matrix_size(module, args.location)
         dtype = module.weight.dtype
         device = get_offloaded_device(module)
+        exec_device = get_execution_device(module)
 
-        weight = self.weights[size, dtype, device]
+        weight = self.weights.get(size, dtype, device, construct_device=exec_device)
```

Comment:
So we're storing the hadamard based on the size, dtype, the offloaded device, and execution device?

Would this mean the same hadamard (e.g hadamard 512) could be in the dictionary if on two different execution devices?

Reply:
Isn't __missing__ called when `get()` is called, in which case the tuple of constructor arguments are also acting as keys?

Based on the docstring/implementation of __missing__ here: https://github.com/neuralmagic/compressed-tensors/blob/98a0cd7ed4a26f504e26fe04be80fb5109e575e8/src/compressed_tensors/utils/helpers.py#L367